go-mode.el

```diff
@@ -3051,6 +3051,45 @@ This handles multi-line comments with a * prefix on each line."
 
 
 
+;; Convenient go-* functions for gopls features available through code
+;; actions, that work across LSP clients:
+
+(defun go-mode--code-action (kind)
+  "Request and invoke the specified kind of code actions for the current selection."
+  (cond
+   ((and (boundp 'eglot--managed-mode) eglot--managed-mode)
+    (let ((beg-end (eglot--code-action-bounds)))
+      (eglot-code-actions (car beg-end) (cadr beg-end) kind t)))
+   ((and (boundp 'lsp-mode) lsp-mode)
+    (lsp-execute-code-action-by-kind kind))
+   (error "buffer is not managed by a known LSP client")))
+
+(defun go-freesymbols ()
+  "View free symbols referenced by the current selection. Requires gopls v0.16."
+  (interactive)
+  (go-mode--code-action "source.freesymbols"))
+
+(defun go-doc ()
```

Comment:
Something like `go-browse-doc` might fit better, both to distinguish it from the existing `godoc` and to make it more apparent that it'll open the documentation in a browser. WDYT?

Reply:
> Something like go-browse-doc might fit better, both to distinguish it from the existing godoc and to make it more apparent that it'll open the documentation in a browser. WDYT?

My preference was to omit it for the sake of brevity, but I confess I wasn't aware of the existing `godoc` function, so I now see the potential for confusion. Done.